ansible/provision.yaml

```diff
@@ -0,0 +1,120 @@
+---
+- name: Provision a k8s stack
+  hosts: localhost
+  connection: local
+  
+  # Include the aws account specific cluster variables
+  vars_files:
+    - vars/{{ environment_type }}-{{ aws_region }}.yaml
+    - vaults/{{ platform }}/vault_{{ environment_type }}.yaml
+
+  # Include the environment specific cluster variables
+  pre_tasks:
+    - include_vars: vars/{{ platform }}-{{ environment_type }}-{{ aws_region }}.yaml
+      when: (cluster_environment == "")
+
+    - include_vars: vars/{{ platform }}-{{ environment_type }}-{{ cluster_environment }}-{{ aws_region }}.yaml
+      when: (cluster_environment != "")
+
+  tasks:
+    - name: Set the stack name when cluster_environment variable is set
+      set_fact: stack_name='{{ platform }}-{{ cluster_name }}-{{ cluster_environment }}-{{ aws_region }}'
+      when: (cluster_environment != "")
+
+    - name: Set the stack name when cluster_environment variable is not set
+      set_fact: stack_name='{{ platform }}-{{ cluster_name }}-{{ aws_region }}'
+      when: (cluster_environment == "")
+
+    - name: Log vars
+      debug:
+        msg: "Stack Name: '{{ stack_name }}'"
+
+    # Create the cluster.yaml file for provisioning
+    - name: Generate the cluster config
+      template:
+        src: templates/cluster-template.yaml.j2
+        dest: /ansible/cluster.yaml
+
+    # If credentials are being shared between environments, generate certs except ca
+    - name: Generate cluster credentials if necessary
+      command: "{{ item }}"
+      with_items:
+        - find credentials/ ! -name 'ca*' -type f -exec rm -f {} +
+        - kube-aws render credentials
+      when: (share_cluster_credentials | lower == "y")
+
+    # If credentials are not being shared between environments, generate certs
+    - name: Generate cluster credentials if necessary
+      command: "{{ item }}"
+      with_items:
+        - rm -rf credentials/*
+        - kube-aws render credentials --generate-ca
+      when: (share_cluster_credentials | lower != "y")
+    
+    # Set the S3 bucket where the stack config is going to be stored
+    - name: Set the provisioner bucket
+      set_fact:
+        provisioner_bucket: "{{ 'k8s-provisioner-test-{{ aws_region }}' if (environment_type == 'd') else 'k8s-provisioner-prod-{{ aws_region }}' }}"
```

Comment:
Maybe we can include also the creation of this bucket in the provisioner. But we can do this at a later time, this is just as a reminder.

Reply:
yes Sorin. Will add a card to the board